p2p/simulations/http.go

```diff
@@ -381,6 +395,37 @@ func (s *Server) StreamNetworkEvents(w http.ResponseWriter, req *http.Request) {
 	}
 }
 
+// CreateSnapshot creates a network snapshot
+func (s *Server) CreateSnapshot(w http.ResponseWriter, req *http.Request) {
+	network := req.Context().Value("network").(*Network)
+
+	snap, err := network.Snapshot()
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	s.JSON(w, http.StatusOK, snap)
+}
+
+// LoadSnapshot loads a snapshot into a network
+func (s *Server) LoadSnapshot(w http.ResponseWriter, req *http.Request) {
+	network := req.Context().Value("network").(*Network)
+
+	snap := &Snapshot{}
+	if err := json.NewDecoder(req.Body).Decode(snap); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	if err := network.Load(snap); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	s.JSON(w, http.StatusOK, network)
```

Comment:
We have loaded a snapshot into the network and produced a new state of the network, so here we are returning that new network state.